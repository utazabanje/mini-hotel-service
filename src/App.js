import React, { Component } from 'react';
import './App.css';

class App extends Component {

  onAdd = (a, b) => {
    return a + b
  }


  render() {
    return (
      <div className="App">
      
      </div>
    )
  }
}

export default App;
