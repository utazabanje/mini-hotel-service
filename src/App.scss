@import './assets/scss/colors';
@import './assets/scss/responsive_media';

body {
  background-color: $main-color;

  input[type=number]::-webkit-inner-spin-button, 
  input[type=number]::-webkit-outer-spin-button { 
    -webkit-appearance: none; 
    margin: 0; 
  }

  .btn.btn-light {
    background-color: $second-color;
    color: $white-website-color;
    border-color: $second-color;

    &:hover {
      background-color: $white-website-color;
      color: $black-website-color;
      border-color: $main-color;
    }
  }

  .clearfix {
    clear: both;
  }

  .navigation-bar {
    background-color: $main-color;
    border-bottom: 1px solid $second-color;
    
    .navbar-brand {
      color: #fff;
    }

    .navbar-collapse {

      .navbar-nav {
        text-align: center;

        .nav-item {

          .nav-link {
            color: $white-website-color;
    
            &:hover {
              color: $black-website-color;
            }
          }
        }
      }
    }
  }

  // WELCOME PAGE START
  .welcome-page {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;

      .welcome-title {
        color: $gray-website-color;
      }

      .welcome-subtitle {
        color: $black-website-color;
      }
  }
  // WELCOME PAGE END

  // STAFF PAGE START
  .staff-login-page {
    

    .staff-login-page-title {
      margin-top: 30px;
      text-align: center;
    }

    .staff-login-page-subtitle {
      text-align: center;
    }

    .staff-login-form {
      padding-top: 50px;
      width: 100%;
      max-width: 500px;
      margin: 0 auto;
    }
  }
  // STAFF PAGE END

  .checkin-page {
    
    .checkin-page-title {
      text-align: center;
      margin: 30px 0;
    }

    form {
      padding: 30px 0;
    }

    @media screen and (max-width: $mobile-size) {
      form {
        padding-top: 0;
      }
    }
  }

  // MANAGE GUESTS PAGE START
  .manage-guests-page {
    position: relative;
    height: 100vh;

    .datepicker-title {
      margin-bottom: 0;
      padding-left: 20px;
    }

    .react-datepicker-wrapper {
      width: 100%;
      margin-bottom: 20px;
      padding-left: 20px;
    }

    .manage-guests-title {
      text-align: center;
      margin: 30px 0;
    }

    .btn.btn-light {
      display: block;
      margin: auto;
      margin-top: 20px;
    }

    .spinner-border {
      border-width: 8px;
      position: absolute;
      top: 35%;
      left: 45%;
    }

    .text-light {
      width: 100px;
      height: 100px;
    }

    .card-wrapper {

      .card {
        width: calc(50% - 40px);
        margin: 20px;
        float: left;

        .btn.btn-danger {
          position: absolute;
          top: 2px;
          right: 2px;
        }

        .card-text {
          font-size: 13px;
          color: gray;

          span {
            display: block;
            font-size: 15px;
            font-weight: 500;
            color: $black-website-color
          }
        }
      }

      @media screen and (max-width: $mobile-size) {
      
        .card {
          width: 100%;
          float: none;
          margin: 0;
          margin-bottom: 20px;
        }
      }
    }

    @media screen and (max-width: $mobile-size) {
      
      .datepicker-title {
        padding-left: 0;
      }
      .react-datepicker-wrapper {
        padding-left: 0;
      }
    }
  }
  // MANAGE GUESTS PAGE END
}
