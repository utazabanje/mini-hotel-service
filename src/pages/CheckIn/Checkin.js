import React, { Component } from "react";
import { Form, Col, Button } from 'react-bootstrap';
import Fire from "../../config/FirebaseLogin";

class CheckIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      guests: [],
      currentName: '',
      currentLastName: '',
      currentEmail: '',
      currentAddress: '',
      currentCity: '',
      currentDate: Date.now(),
      currentZip: '',
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    });
  }
 
  handleSubmit = (e) => {
    e.preventDefault();

    const guestsRef = Fire.database().ref('guests');
    const guest = {
      name: this.state.currentName,
      lastname: this.state.currentLastName,
      email: this.state.currentEmail,
      address: this.state.currentAddress,
      city: this.state.currentCity,
      zip: this.state.currentZip,
      date: this.state.currentDate,
    }
    guestsRef.push(guest);
    
    this.setState({
      currentName: '',
      currentLastName: '',
      currentEmail: '',
      currentAddress: '',
      currentCity: '',
      currentDate: Date.now(),
      currentZip: '',
    });
  }

  render() {
    return (
      <div className="checkin-page">
        <h1 className="checkin-page-title">Check In</h1>

        <Form className="checkin-login-form"
          onSubmit={this.handleSubmit}>
          <Form.Row>
            <Form.Group as={Col} sm="12" md="6" controlId="validateName">
              <Form.Label>First Name</Form.Label>
              <Form.Control value={this.state.currentName} onChange={this.handleChange} name="currentName" type="text" placeholder="First Name" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your name.
              </Form.Control.Feedback>
            </Form.Group>

            <Form.Group as={Col} sm="12" md="6" controlId="validateLastName">
              <Form.Label>Last Name</Form.Label>
              <Form.Control value={this.state.currentLastName} onChange={this.handleChange} name="currentLastName" type="text" placeholder="Last Name" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your last name.
              </Form.Control.Feedback>
            </Form.Group>
          </Form.Row>

          <Form.Row>
            <Form.Group as={Col} sm="12" md="6" controlId="validateEmail">
              <Form.Label>Email</Form.Label>
              <Form.Control value={this.state.currentEmail} onChange={this.handleChange} name="currentEmail" type="email" placeholder="Enter email" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your email.
              </Form.Control.Feedback>
            </Form.Group>

            <Form.Group as={Col} sm="12" md="6" controlId="validateAddress">
              <Form.Label>Address</Form.Label>
              <Form.Control value={this.state.currentAddress} onChange={this.handleChange} name="currentAddress" type="text" placeholder="1234 Main St" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your address.
              </Form.Control.Feedback>
            </Form.Group>
          </Form.Row>

          <Form.Row>
            <Form.Group as={Col} sm="12" md="6" controlId="validateCity">
              <Form.Label>City</Form.Label>
              <Form.Control value={this.state.currentCity} onChange={this.handleChange} name="currentCity" type="text" placeholder="City" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your city.
              </Form.Control.Feedback>
            </Form.Group>

            <Form.Group as={Col} sm="12" md="6" controlId="validateZip">
              <Form.Label>Zip</Form.Label>
              <Form.Control value={this.state.currentZip} onChange={this.handleChange} name="currentZip" type="number" placeholder="zip" required/>
              <Form.Control.Feedback type="invalid">
                Please enter your zip code.
              </Form.Control.Feedback>
            </Form.Group>
          </Form.Row>

          <Button variant="light" type="submit">
            Submit
          </Button>
        </Form>
      </div>
    );
  }
}

export default CheckIn;
