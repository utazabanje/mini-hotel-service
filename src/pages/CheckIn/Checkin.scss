@import './assets/scss/colors';
@import './assets/scss/responsive_media';

.checkin-page {
    
  .checkin-page-title {
    text-align: center;
    margin: 30px 0;
    color: $green-website-color;
  }

  form {
    padding: 30px 0;
  }

  @media screen and (max-width: $mobile-size) {
    form {
      padding-top: 0;
    }
  }
}