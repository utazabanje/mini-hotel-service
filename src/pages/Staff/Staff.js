import React, { Component } from "react";
import StaffLogin from "../../components/StaffLogin/StaffLogin";
import ManageGuest from "../../components/ManageGuests/ManageGuests";
import Fire from "../../config/FirebaseLogin";

class Staff extends Component {
	constructor(props) {
		super(props);

		this.state = {
			user: {}
		};
	}

	componentDidMount() {
    this.authListener();
  }

	authListener = () => {
    Fire.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({ user });
      } else {
        this.setState({ user: null })
      }
    })
  }

	render() {
		return (
			<div className="staff-page">
				{this.state.user ? <ManageGuest /> : <StaffLogin />}
			</div>
		);
	}
}

export default Staff;
