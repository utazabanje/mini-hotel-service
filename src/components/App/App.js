import React, { Component } from 'react';
import './App.scss';

class App extends Component {

  onAdd = (a, b) => {
    return a + b
  }


  render() {
    return (
      <div className="App">
      
      </div>
    )
  }
}

export default App;
