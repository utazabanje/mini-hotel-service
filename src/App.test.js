import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ManageGuest from './components/ManageGuests/ManageGuests';
import Staff from './pages/Staff/Staff';
import CheckIn from './pages/CheckIn/Checkin';

configure({adapter: new Adapter()});

describe('Manage Guest component with guests as empty string', () => {
  it('should return 0', () => {
    const wrapper = shallow(<ManageGuest />);
    const guestState = wrapper.state().guests;
    expect(guestState.length).toEqual(0);
  });
});

describe('Manage Guest snapshot', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<ManageGuest />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});

describe('Staff component should render Manage component if user is signed in', () => {
  it('One Manage Component', () => {
    const wrapper = shallow(<Staff />);
    const numOfComponents = wrapper.find('ManageGuest');
    
    expect(numOfComponents.length).toEqual(1);
  });
});

describe('Check In title', () => {
  it('Check In', () => {
    const wrapper = shallow(<CheckIn />);
    const text = wrapper.find('h1.checkin-page-title').text();
    
    expect(text).toBe('Check In');
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
