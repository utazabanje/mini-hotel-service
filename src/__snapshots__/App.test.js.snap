// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Guest snapshot Snapshot 1`] = `
<div
  className="manage-guests-page"
>
  <button
    className="btn btn-light"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Log Out
  </button>
  <h1
    className="manage-guests-title"
  >
    Manage guest page
  </h1>
  <div
    className="spinner-border text-light"
  />
  <p
    class="datepicker-title"
  >
    Pick a date:
  </p>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value="04/27/2019"
      />
    </div>
  </div>
  <div
    className="card-wrapper"
  >
    <div
      className="clearfix"
    />
  </div>
</div>
`;
